File: javaQuarkusShip/src/main/java/pl/battleships/javaquarkusship/client/RestClientGameApi.java
package pl.battleships.javaquarkusship.client;

import io.swagger.annotations.ApiParam;
import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
import pl.battleships.javaquarkusship.api.dto.PositionDto;
import pl.battleships.javaquarkusship.api.dto.ShotStatusDto;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import javax.ws.rs.*;

@Path("/game")
@RegisterRestClient(configKey = "game")
public interface RestClientGameApi {

    @POST
    @Path("/{id}/shot")
    ShotStatusDto shot(@PathParam("id") @ApiParam("game id") String id, @Valid @NotNull PositionDto positionDto);
}

File: kotlinQuarkusShip/src/main/kotlin/pl/battleships/kotlinquarkusship/client/RestClientGameApi.kt
package pl.battleships.kotlinquarkusship.client

import org.eclipse.microprofile.rest.client.inject.RegisterRestClient
import pl.battleships.api.dto.PositionDto
import pl.battleships.api.dto.ShotStatusDto
import javax.validation.Valid
import javax.validation.constraints.NotNull
import javax.ws.rs.POST
import javax.ws.rs.Path
import javax.ws.rs.PathParam


@Path("/game")
@RegisterRestClient(configKey = "game")
interface RestClientGameApi {

    @POST
    @Path("/{id}/shot")
    fun shot(@PathParam("id") id: String, @Valid @NotNull positionDto: PositionDto): ShotStatusDto
}